src/kernel/mconfig.ml

```diff
@@ -341,13 +366,14 @@ let merlin_flags = [
 let query_flags = [
   (
     "-verbosity",
-    Marg.param "integer" (fun verbosity query ->
+    Marg.param Verbosity.param_spec (fun verbosity query ->
         let verbosity =
-          try int_of_string verbosity
-          with _ -> invalid_arg "argument should be an integer"
+          Verbosity.of_string verbosity        
         in
         {query with verbosity}),
-    "<integer> Verbosity determines the number of expansions of aliases in answers"
+    "\"smart\" | <integer> Verbosity determines the number of \
+      expansions of aliases in answers. \"smart\" automatically \
+      determines this number itself."
```

Comment:
In fact, `smart` has a precise behavior: it expands Module types and nothing else. The spec should say that instead of a blurry statement.